oracle-uri-proposal.md

```diff
@@ -0,0 +1,196 @@
+
+Oracles must announce their intention to attest to the outcome of an event ahead of time.
+This is done by signing a message which includes the event id.
+Additionally it usually involves committing to some auxiliary cryptographic data (e.g. nonces) which allows the clients to _anticipate_ the final signature for any message.
+The announcement signature is made by suffixing the event id with `!` and appending the cryptographic data.
+
+_how to encode the data is left for a future revision_
+
+### Attestation
+
+When attesting to a particular outcome for the event the oracle appends `=` and the outcome text onto the event id and signs the result.
+For example, if the outcome of `NBA/s2019/2020-08-11/PHI_PHX?vs` is `PHX_win` then the oracle signs `NBA/s2019/2020-08-11/PHI_PHX?vs=PHX_win`.
+
+### Source Identifiers
+
+If the first segment of a path contains an `@` symbol then the text that precedes it is a _source identifier_.
+The source identifier describes a third party that the oracle is using to decide the outcome.
+This allows an oracle to explicitly state that it is vulnerable to manipulation from this party and therefore abdicate part of its usual responsibility.
+The oracle is conveying that it may blindly attest to whatever the source tells it without verifying it. 
+
+An oracle who does not provide a source identifier is asserting that it will take the measurement of the event itself or otherwise verify it somehow before attesting to it. 
+It is completely acceptable from the point of view of this specification for an oracle to omit a source identifier even if in reality it is only using one online source to decide the outcome; it is simply a question of whether the oracle finds it advantageous to inform its users of this fact.
+
+Clients should interpret the event that the event id is describing as if the source identifier were absent.
+
+Examples:
+
+- `espn.com@NBA/s2019/2020-08-11/PHI_PHX?vs` indicates the oracle is using `espn.com` to determine the outcome of the `NBA` match (it is not watching the match itself).
+- `wttr.in@Earth/Paris/2020-08-19?weather` indicates that the oracle is using `wttr.in` to determine the weather in Paris on the date (it is not using its own weather satellite).
+
+### Event Types
+
+The _event type_ occupies the _query string_ part of the URI.
+Event types will be introduced in their own separate RFCs and all event types mentioned here are just an example.
+The event type is a kebab-case identifier which may be followed by _parameters_ enclosed in parenthesis.
+The parameters are a list of comma separated values.
+Each event type defines its parameters and together they determine the outcomes for the event.
+
+The event type will usually define how to interpret the preceding path segments.
+For example, for a `vs` event type may require that the final path segment be in the form `<team1>_<team2>` or a `weather` event type may require the last two segments to be in the form `<location>/<date>`.
```

Comment:
I think it may be beneficial to add a version to the URI to make things more extensible

Reply:
Sorry I missed this comment.
In my experience, versions in general make things less extensible because as soon as a client sees a version it does not understand it fails to parse whatever comes next. What kind of extensibility do you have in mind?

Reply:
Related to #88